github/users_packages.go

```diff
@@ -0,0 +1,193 @@
+	var u string
+	if user != "" {
+		u = fmt.Sprintf("users/%v/packages/%v/%v", user, packageType, packageName)
+	} else {
+		u = fmt.Sprintf("user/packages/%v/%v", packageType, packageName)
+	}
+	req, err := s.client.NewRequest("DELETE", u, nil)
+	if err != nil {
+		return nil, err
+	}
+	return s.client.Do(ctx, req, nil)
+
+}
+
+// Restore a package to a user. Passing the empty string will
+// restore the package for the authenticated user.
+//
+// GitHub API docs: https://docs.github.com/en/rest/reference/packages#restore-a-package-for-the-authenticated-user
+// GitHub API docs: https://docs.github.com/en/rest/reference/packages#restore-a-package-for-a-user
+func (s *UsersService) RestorePackage(ctx context.Context, user, packageType, packageName string) (*Response, error) {
+	var u string
+	if user != "" {
+		u = fmt.Sprintf("users/%v/packages/%v/%v/restore", user, packageType, packageName)
+	} else {
+		u = fmt.Sprintf("user/packages/%v/%v/restore", packageType, packageName)
+	}
+	req, err := s.client.NewRequest("POST", u, nil)
+	if err != nil {
+		return nil, err
+	}
+	return s.client.Do(ctx, req, nil)
+
+}
+
+// Get all versions of a package for a user. Passing the empty string will
+// get versions for the authenticated user.
+//
+// GitHub API docs: https://docs.github.com/en/rest/reference/packages#get-all-package-versions-for-a-package-owned-by-the-authenticated-user
+// GitHub API docs: https://docs.github.com/en/rest/reference/users#delete-an-email-address-for-the-authenticated-user
+func (s *UsersService) PackageGetAllVersions(ctx context.Context, user, packageType, packageName string) ([]*PackageVersion, *Response, error) {
+	var u string
+	if user != "" {
+		u = fmt.Sprintf("users/%v/packages/%v/%v/versions", user, packageType, packageName)
+	} else {
+		u = fmt.Sprintf("user/packages/%v/%v/versions", packageType, packageName)
+	}
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	var versions []*PackageVersion
+	resp, err := s.client.Do(ctx, req, &versions)
+	if err != nil {
+		return nil, resp, err
+	}
+	return versions, resp, nil
+
+}
+
+// Get a specific version of a package in for a user. Passing the empty string will
+// get the version for the authenticated user.
+//
+// GitHub API docs: https://docs.github.com/en/rest/reference/packages#get-a-package-version-for-the-authenticated-user
+// GitHub API docs: https://docs.github.com/en/rest/reference/packages#get-a-package-version-for-a-user
+func (s *UsersService) PackageGetVersion(ctx context.Context, user, packageType, packageName string, packageVersionID int64) (*PackageVersion, *Response, error) {
+	var u string
+	if user != "" {
+		u = fmt.Sprintf("users/%v/packages/%v/%v/versions/%v", user, packageType, packageName, packageVersionID)
+	} else {
+		u = fmt.Sprintf("user/packages/%v/%v/versions/%v", packageType, packageName, packageVersionID)
+	}
+	req, err := s.client.NewRequest("GET", u, nil)
```

Comment:
Please add a blank line above this line for consistency with the rest of the repo.